backend/pkg/modules/internal/fuzzer/clients/kubernetes.go

```diff
@@ -98,6 +99,24 @@ func (l *K8sClient) TriggerFuzzingJob(apiID int64, endpoint string, securityItem
 	return nil
 }
 
+func (l *K8sClient) StopFuzzingJob(apiID int64) error {
+	if l.currentJob == nil {
+		return fmt.Errorf("no current k8s job to terminate")
+	}
+	var zero int64 // = 0
+	policy := metav1.DeletePropagationForeground
+	deleteOptions := &metav1.DeleteOptions{
+		GracePeriodSeconds: &zero,
+		PropagationPolicy:  &policy,
+	}
+	err := l.hClient.BatchV1().Jobs(l.currentJob.Namespace).Delete(context.TODO(), l.currentJob.Name, *deleteOptions)
+	l.currentJob = nil
```

Comment:
why clearing `currentJob` here and not after error check?

Reply:
To reset l.currentJob in all case. even if error. But I can put it after if you want, I don't test it for now before the start (if a such test exists, we will have a lock for ever in case of error during the stop)